tools/cldr-apps/src/main/java/org/unicode/cldr/web/SurveyTool.java

```diff
@@ -129,92 +124,112 @@ private void serveBustedPage(HttpServletRequest request, PrintWriter out) {
-            out.write("var survURL = '" + survURL + "';\n");
-            out.write("(document.getElementById('redir') || {}).href = newUrl;\n");
-            out.write("var dstatus = document.getElementById('st_err');\n");
-            out.write("if (dstatus != null) {\n");
-            out.write("  dstatus.appendChild(document.createElement('br'));\n");
-            out.write("  dstatus.appendChild(document.createTextNode('.'));\n");
-            if (useDojo(request)) {
-                out.write("  require(['dojo/ready'], function(ready) {\n");
-                out.write("    ready(function() {\n");
-            } else {
-                out.write("  $(function() {\n"); // jquery
+        if(!useDojo(request)) {
+            out.write("<html lang='" + SurveyMain.TRANS_HINT_LOCALE.toLanguageTag() + "' class='claro'>\n");
+            out.write("<head>\n");
+            out.write("<meta http-equiv='Content-Type' content='text/html; charset=UTF-8'>\n");
+            out.write("<title>CLDR Survey Tool | Starting</title>\n");
+            includeCss(request, out);
+            out.write(VettingViewer.getHeaderStyles() + "\n");
+            try {
+                includeJavaScript(request, out);
+            } catch (JSONException e) {
+                SurveyLog.logException(e, "Including JavaScript");
             }
-            out.write("        dstatus.appendChild(document.createTextNode('.'));\n");
-            out.write("        window.setTimeout(function () {\n");
-            out.write("          dstatus.appendChild(document.createTextNode('.'));\n");
-            out.write("          const load = function (data) {\n");
-            out.write("            dstatus.appendChild(\n");
-            out.write("              document.createTextNode(\n");
-            out.write("                'Loaded ' +\n");
-            out.write("                  data.length +\n");
-            out.write("                  ' bytes from SurveyTool. Reloading this page..'\n");
-            out.write("              )\n");
-            out.write("            );\n");
-            out.write("            window.location.reload(true);\n");
-            out.write("          };\n");
-            out.write("          const xhrArgs = {\n");
-            out.write("            url: survURL,\n");
-            out.write("            load: load,\n");
-            out.write("          };\n");
-            out.write("          cldrAjax.sendXhr(xhrArgs);\n");
-            out.write("        }, 2000); // two seconds\n");
-            out.write("      });\n");
-            if (useDojo(request)) {
-                out.write("  });\n");
+            // fire up the Vue based waiting page
+            out.write("</head>\n<body>\n");
+            writeWaitingNavbarHtml(out);
+            out.write("<div id='app'></div>");
+            out.write("<script>cldrBundle.showWaiting('#app');</script>");
+            out.write("</body>");
```

Comment:
instead of actual js inside html, how about just including an element with a certain id, whose presence tells cldrBundle.start() to show the waiting page instead of the normally-running page?

Reply:
@btangmu a router! Actually, what we can have is one entrypoint, something like `updateVue()` that's _always_ called.  And it will internally have some kind of map like this:

```json
{
  "/v#about": "About.vue",
  "/v#retry": "Waiting.vue"
}
```

Would that work for now? then there would be minimal code change needed for moving something from the "new" JS model to the "vue" JS model, and there'd be no extraneous `cldrBundle.` level exports.

Reply:
That sounds better... "loose coupling" means most of our code shouldn't know or care whether we're using webpack